tools/tests/add_and_update.py

```diff
@@ -0,0 +1,129 @@
+#! /usr/bin/env python3
+
+import sys
+
+from util import *
+from init_tests import init_project
+
+project_name="aou_project"
+project_path=os.path.join("./",project_name)
+
+def expect_subvalue(ll, name, tag, value):
+   """Inside ll, find the list that contains `name` and test the value stored in that list for the given tag against value.
+
+   :param ll  An array of lists.
+   :param name  The name to search for.
+   :tag  The tag that contains the value we to test.
+   :value  The expected value.
+   """
+
+   for i in ll:
+      if i["name"] == name:
+         return i[tag] == value
+   return False
+
+
+def setup_project():
+   """Setup the project ensuring there is a fresh project file.
+   """
+
+   shutil.rmtree(project_path, ignore_errors=True)
+   init_project(project_name, "v1.0.0", project_path)
+   assert(os.path.isdir(project_path))
+
+
+def test_app_compile_options():
+   """Add an application to the project and test its compile options can be updated.
+   """
+
+   out, err = antler_in_proj_cmd(project_path, "add app App1 C++ \\\\-O1")
```

Comment:
The same for `err` in lines:
add_and_update.py:39:8: W0612: Unused variable 'err' (unused-variable)
add_and_update.py:54:8: W0612: Unused variable 'err' (unused-variable)
add_and_update.py:71:8: W0612: Unused variable 'err' (unused-variable)
add_and_update.py:98:8: W0612: Unused variable 'err' (unused-variable)

```suggestion
   out, _ = antler_in_proj_cmd(project_path, "add app App1 C++ \\\\-O1")
```